github/resource_github_organization_settings.go

```diff
@@ -0,0 +1,373 @@
+package github
+
+import (
+	"context"
+	"log"
+	"strconv"
+
+	"github.com/google/go-github/v47/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resouceGithubOrganizationSettings() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubOrganizationSettingsCreateOrUpdate,
+		Read:   resourceGithubOrganizationSettingsRead,
+		Update: resourceGithubOrganizationSettingsCreateOrUpdate,
+		Delete: resourceGithubOrganizationSettingsDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"billing_email": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"company": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"email": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"twitter_username": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"location": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"has_organization_projects": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"has_repository_projects": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  true,
+			},
+			"default_repository_permission": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "read",
+				ValidateFunc: validation.StringInSlice([]string{"read", "write", "admin", "none"}, false),
```

Comment:
If we find ourselves duplicating this logic in the future, we may abstract this into its own helper somewhere.